PC/launcher.c

```diff
@@ -1371,6 +1371,79 @@ get_version_info(wchar_t * version_text, size_t size)
     }
 }
 
+static void
+show_help_text(wchar_t ** argv)
+{
+    wchar_t version_text [MAX_PATH];
+#if defined(_M_X64)
+    BOOL canDo64bit = TRUE;
+#else
+    /* If we are a 32bit process on a 64bit Windows, first hit the 64bit keys. */
+    BOOL canDo64bit = FALSE;
+    IsWow64Process(GetCurrentProcess(), &canDo64bit);
+#endif
+
+    get_version_info(version_text, MAX_PATH);
+    fwprintf(stdout, L"\
+Python Launcher for Windows Version %ls\n\n", version_text);
+    fwprintf(stdout, L"\
+usage:\n\
+%ls [launcher-args] [python-args] script [script-args]\n\n", argv[0]);
+    fputws(L"\
+Launcher arguments:\n\n\
+-2     : Launch the latest Python 2.x version\n\
+-3     : Launch the latest Python 3.x version\n\
+-X.Y   : Launch the specified Python version\n", stdout);
+    if (canDo64bit) {
+        fputws(L"\
+     The above all default to 64 bit if a matching 64 bit python is present.\n\
+-X.Y-32: Launch the specified 32bit Python version\n\
+-X-32  : Launch the latest 32bit Python X version\n\
+-X.Y-64: Launch the specified 64bit Python version\n\
+-X-64  : Launch the latest 64bit Python X version", stdout);
+    }
+    fputws(L"\n-0     : List the available pythons", stdout);
+    fputws(L"\n\nThe following help text is from Python:\n\n", stdout);
+    fflush(stdout);
+}
+
+static void
+show_python_list(wchar_t ** argv)
+{
+    INSTALLED_PYTHON * result = NULL;
+    INSTALLED_PYTHON * ip = installed_pythons;
+    size_t i = 0;
+    wchar_t *fmt = L"\n -%ls-%d\t%ls";
+
+    /*
+    * Output informational messages to stderr to keep output
+    * clean for use in pipes, etc.
+    */
+    fwprintf(stderr,
+             L"Installed Pythons found by %s Launcher for Windows", argv[0]);
+
+    if (num_installed_pythons == 0)
+        locate_all_pythons();
+
+    if (num_installed_pythons == 0)
+        fwprintf(stderr, L"\nNo Installed Pythons Found!");
+    else
+    {
+        for (i = 0; i < num_installed_pythons; i++, ip++) {
+            fwprintf(stdout, fmt, ip->version, ip->bits, ip->executable);
```

Comment:
Does is really need to print the executable path? It's trivial to get that information manually.